ansible-ipi-install/roles/node-prep/tasks/100_power_off_cluster_servers.yml

```diff
@@ -24,4 +24,5 @@
       retries: 10
       delay: 5
       with_items: "{{ groups['poweroff_hosts'] }}"
+      ignore_errors: true
```

Comment:
we don't want to ignore because if we ignore, its possible it will mess up the entire install. It's better it fail where it fails.

Reply:
Should then include some clause of it not being empty? because actual status is that list is empty and instead of ignoring it, it tries to use it and fails (groups['poweroff_hosts'])

In python when you do:

~~~py
var=[1,2,3]

for item in var:
    print(item)

# or
var = []
for item in var:
    print(item)

~~~

In both cases, it works, as first it loops over three elements and second it loops over 0 elements.

In this case it reacts more like a dictionary var with undefined element and it fails completely

Reply:
I've updated the PR to check if the group is defined and if the number of elements is higher than 0, to simulate above Python behavior

Reply:
Yes. I think checking if the group is defined is the way to go /lgtm